excretor/src/routing.rs

```diff
@@ -63,10 +71,59 @@ mod handlers {
         per_page: u32,
     }
 
+    #[derive(Deserialize)]
+    pub struct AuthCallback {
+        code: String,
+    }
+
+    #[derive(Deserialize)]
+    pub struct AuthToken {
+        #[serde(default = "String::new")]
```

Comment:
You can make its type `Option<String>` instead of setting a default empty value and using `is_empty`.